variables.tf

```diff
@@ -41,4 +42,45 @@ variable "terraform_state_config" {
     storage_account_name = string
     backend_service_connection = string
   })
+  description = "The configuration of the Terraform state. The state will be saved in the given storage account ind the DevOps subscription using the backend service connection."
+}
+
+variable "stage" {
+  type = string
+  description = "Name of the current stage"
+}
+
+variable "location" {
+  type = string
+  description = "The default location used for resources in this Landing Zone."
+}
+
+variable "vnet_config" {
+  type = object({
+    dns_server = list(string)
+    address_space   = string
+    subnets = map(string) 
+    # doku was key
+  })
+  description = <<-DOC
+  ```
+  If you want to provide a virtual network, please provide the following values: 
+    dns_server: DNS Servers that will be used in the network.
+    address_space: Address space of the virtual network in CIDR notation.
+    subnets: Subnets that will be created in the virtual network. Use 'Usecase' as the key and the address prefix as the value in CIDR notation.    
+  ```
+  DOC
+  default = null
+}
+
+variable "skip_provider_registration" {
+  type        = bool
+  description = "Allows you to skip the provider registration when initilizing the azurerm provider. This is useful in development environments where not every provider can be registered."
+  default     = false
+}
+
+variable "virtual_machine_template" {

```

Comment:
Instead of a boolean we could prefill the template already with values and leave it out if nothing was supplied

Reply:
ad discussed we leave the module like that